ddspipe_core/src/cpp/communication/dds/DdsBridge.cpp

```diff
@@ -97,187 +104,185 @@ void DdsBridge::disable() noexcept
     }
 }
 
-void DdsBridge::create_all_tracks_()
+void DdsBridge::create_endpoint(
+        const ParticipantId& participant_id,
+        const EndpointKind& discovered_endpoint_kind)
 {
     std::lock_guard<std::mutex> lock(mutex_);
 
-    const auto& ids = participants_->get_participants_ids();
+    switch (discovered_endpoint_kind)
+    {
+        case EndpointKind::reader:
+            create_writer_and_its_tracks_nts_(participant_id);
+            break;
+        case EndpointKind::writer:
+            create_reader_and_its_track_nts_(participant_id);
+            break;
+        default:
+            logError(DDSPIPE_DDSBRIDGE, "Invalid kind " << discovered_endpoint_kind << " to create an endpoint.");
+            break;
+    }
+}
 
-    // Figure out what writers need to be created
-    std::set<ParticipantId> writers_to_create;
+void DdsBridge::remove_endpoint(
+        const ParticipantId& participant_id,
+        const EndpointKind& removed_endpoint_kind)
+{
+    std::lock_guard<std::mutex> lock(mutex_);
 
-    for (const ParticipantId& id : ids)
+    switch (removed_endpoint_kind)
     {
-        const auto& routes_it = routes_.find(id);
+        case EndpointKind::reader:
+            remove_writer_and_its_tracks_nts_(participant_id);
+            break;
+        case EndpointKind::writer:
+            remove_reader_and_its_track_nts_(participant_id);
+            break;
+        default:
+            logError(DDSPIPE_DDSBRIDGE, "Invalid kind " << removed_endpoint_kind << " to remove an endpoint.");
+            break;
+    }
+}
 
-        if (routes_it != routes_.end())
-        {
-            // The reader has a route. Create only the writers in the route.
+void DdsBridge::create_writer_and_its_tracks_nts_(
+        const ParticipantId& participant_id)
+{
+    assert(participant_id != DEFAULT_PARTICIPANT_ID);
+
+    // Create the writer
+    auto writer = create_writer_nts_(participant_id);
```

Comment:
Another corner case: shouldn't we avoid creating writers if not present in any route?

Reply:
Related: should we also avoid creating a reader if there is a "blockage" route?